dashboard/lib/services/lti/account_unlinker.rb

```diff
@@ -0,0 +1,57 @@
+require 'metrics/events'
+require 'policies/lti'
+
+module Services
+  module Lti
+    class AccountUnlinker < Services::Base
+      attr_reader :user, :auth_option, :issuer, :client_id, :sub
+
+      def initialize(user:, auth_option:)
+        @user = user
+        @auth_option = auth_option
+        auth_id_components = auth_option.authentication_id.split('|')
+        @issuer = auth_id_components[0]
+        @client_id = auth_id_components[1]
+        @sub = auth_id_components[2]
+      end
+
+      def call
+        return unless user.authentication_options.include?(auth_option)
+
+        ActiveRecord::Base.transaction do
+          raise Exception.new('Cannot unlink user\'s only authentication option') if Policies::Lti.only_lti_auth?(user)
+          remove_lti_user_identity
+          auth_option.destroy!
+          promote_coteachers if user.teacher?
+        end
+
+        Metrics::Events.log_event(
+          user: user,
+          event_name: 'lti_account_unlinked',
+          metadata: {
+            lms_name: lti_integration.platform_name,
+            user_type: user.user_type,
+          }
+        )
+      end
+
+      private def lti_integration
+        @lti_integration ||= LtiIntegration.find_by(issuer: issuer, client_id: client_id)
```

Comment:
Super nit: not even sure you'd want to use it here, but we do have `Queries::Lti.get_lti_integration(issuer, client_id)`. Not sure using it here is really a net improvement over what you have.